middle_end/flambda2/terms/flambda.mli

```diff
@@ -106,6 +106,7 @@ module Invalid : sig
     | Apply_cont_of_unreachable_continuation of Continuation.t
     | Defining_expr_of_let of Bound_pattern.t * named
     | Closure_type_was_invalid of Apply_expr.t
+    | Closure_type_was_invalid_classic_mode of Debuginfo.t
```

Comment:
Could we call this `Closure_type_was_invalid_in_lambda` or some such? This happens equally well with classic mode on or off, I imagine?

Reply:
No, this one is only used in classic mode.  I would have reused `Closure_type_was_invalid` but we don't have an `Apply_expr.t` to hand.

Reply:
Ah no, as discussed this is actually used in non-classic mode as well.  I've renamed this constructor, and also added another one, to distinguish (when using Simplify) this case from other cases where closure types are invalid.